pbs_light.go

```diff
@@ -397,6 +399,33 @@ func auction(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 	mRequestTimer.UpdateSince(pbs_req.Start)
 }
 
+// checkForValidBidSize goes through list of bids & find those which are banner mediaType and with height or width not defined
+// determine the num of ad unit sizes that were used in corresponding bid request
+// if num_adunit_sizes == 1, assign the height and/or width to bid's height/width
+// if num_adunit_sizes > 1, reject the bid (remove from list) and return an error
+// return updated bid list object for next steps in auction
+func checkForValidBidSize(bids pbs.PBSBidSlice, bidder *pbs.PBSBidder) pbs.PBSBidSlice {
+	var finalValidBids pbs.PBSBidSlice
```

Comment:
I think this should be initialized with an initial capacity equal to `len(bids)`.

Otherwise Go picks some undocumented initial capacity, and all those `append` calls may or may not cause it to resize the array & copy new elements, which is expensive.

Check out [Go slices in detail](https://blog.golang.org/go-slices-usage-and-internals) for more details.